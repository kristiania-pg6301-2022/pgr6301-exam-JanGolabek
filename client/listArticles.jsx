import React from "react";
import { useLoading } from "./useLoading";

function ArticleCard({ article: { category, title, text, author } }) {
    return (
        <>
            <p>{category}</p>
            <h2>{title}</h2>
            <p>{text}</p>
            <p>
                <i>Written by: {author}</i>
            </p>
        </>
    );
}
export function ListArticles({ listArticles }) {
    const { loading, error, data } = useLoading(listArticles);

    if (loading) {
        return <div>Loading...</div>;
    }
    if (error) {
        return (
            <div>
                <h1>Error</h1>
                <div>{error.toString()}</div>
            </div>
        );
    }
    return (
        <div>
            <h1>Article in the database</h1>
            {data.map((article) => (
                <ArticleCard key={article.title} article={article} />
            ))}
        </div>
    );
}