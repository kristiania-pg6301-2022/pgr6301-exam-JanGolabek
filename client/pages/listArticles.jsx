import React, { useContext } from "react";
import { useLoading } from "../useLoading";
import { ArticlesApiContext } from "../articlesApiContext";

function ArticleCard({ article: { category, title, author, text } }) {
    return (
        <>
            <p>{category}</p>
            <h2>{title}</h2>
            <p>
                <i>Written by: {author}</i>
            </p>
            <p>{text}</p>

        </>
    );
}
export function ListArticles() {
    const { listArticles } = useContext(ArticlesApiContext);
    const { loading, error, data } = useLoading(listArticles);

    if (loading) {
        return <div>Loading...</div>;
    }
    if (error) {
        return (
            <div>
                <h1>Error</h1>
                <div>{error.toString()}</div>
                <div id="error-text">{error.toString()}</div>
            </div>
        );
    }
    return (
        <div>
            <h1>Article in the database</h1>
            {data.map((article) => (
                <ArticleCard key={article.title} article={article} />
            ))}
        </div>
    );
}