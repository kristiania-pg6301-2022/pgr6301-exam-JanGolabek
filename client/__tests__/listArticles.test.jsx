import { ListArticles } from "../listArticles";

import React from "react";
import ReactDOM from "react-dom";

describe("ListArticles component", () => {
    it("shows loading screen", () => {
        const domElement = document.createElement("div");
        ReactDOM.render(<ListArticles />, domElement);
        expect(domElement.innerHTML).toMatchSnapshot();
    });
});