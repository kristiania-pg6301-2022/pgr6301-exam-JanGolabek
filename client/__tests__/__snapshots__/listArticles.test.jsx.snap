// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListArticles component shows articles 1`] = `"<div><h1>Articles in the database</h1><div><form><label>Category:<input id=\\"category-query\\" value=\\"\\"><button>Filter</button></label></form></div><p></p><h2>article 1</h2><p><i>Written by: </i></p><p></p><p></p><h2>article 2</h2><p><i>Written by: </i></p><p></p></div>"`;

exports[`ListArticles component shows error message 1`] = `"<div><h1>Error</h1><div>Error: Something went wrong</div><div id=\\"error-text\\">Error: Something went wrong</div></div>"`;

exports[`ListArticles component shows loading screen 1`] = `"<div>Loading...</div>"`;
